### PR TITLE
Add TF_object_detection

### PR DESCRIPTION
**Pull request by adbeda**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/tf_object_detection/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - tensorflow>=1.12.0
+    - numpy >=1.16.1
+    - protobuf >=3.0.0
+    - jupyter>=1.0.0
```
Comment (scopatz) [inline recipes/tf_object_detection/meta.yaml]: These need to have a single space between them, ie `jupyter >=1.0.0`.  same for tensorflow

File: recipes/tf_object_detection/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  sha256: 476eaf754c61c5afce39e36683bb4a7a6b7f9084e2bc89f075d58e294da27768
+
+build:
+  number: 0
+  skip: true  # [win]  # Skip all windows for now due to numpy/scipy version pinning issues
+  skip: True  # [win32]
+  skip: True  # [win64 and py<35]
+  skip: True  # [osx and py34]
+  skip: True  # [py26]
+  skip: True  # [py33]
```
Comment (scopatz) [inline recipes/tf_object_detection/meta.yaml]: We don't build on python 2.6, 3.3, 3.4, or 3.5.  So these skips are not needed

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: thanks. I used the `tf-feedstock` recipe as an example which have these skips. I'will remove them.

File: recipes/tf_object_detection/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+source:
+  url: https://github.com/Maxar-Corp/ratpac/releases/download/od-{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 476eaf754c61c5afce39e36683bb4a7a6b7f9084e2bc89f075d58e294da27768
+
+build:
+  number: 0
+  skip: True  # [win]  # Skip all windows for now due to numpy/scipy version pinning issues
+  skip: True  # [win32]
```
Comment (scopatz) [inline recipes/tf_object_detection/meta.yaml]: Also, we don't build win32 and so this is redundant with the above

File: recipes/tf_object_detection/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - protobuf >=3.0.0
+    - jupyter >=1.0.0
+    - lxml
+    - matplotlib
+    - cython
+    - contextlib2
+
+test:
+  imports:
+    - tf_object_detection  # [win]
```
Comment (scopatz) [inline recipes/tf_object_detection/meta.yaml]: you are skipping win, and we should have a test for linux and mac.  Or maybe this `# [win]` selector shouldn't be here?

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: oh yeah. it should be removed!

File: recipes/tf_object_detection/meta.yaml (hunk 3 above)
Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @scopatz on the recipe for [tensorflow-feedstock](https://github.com/conda-forge/tensorflow-feedstock/blob/master/recipe/meta.yaml), testing on `linux` and `OSX` has been is indicated to be challenging. The comments section says:

       Skip the import test on Linux as wheel file require a more recent
       version of GLIBC++ than the VM used to build and test package.
       Also skip on OSX due to OSX version requirements, see
       https://github.com/conda-forge/tensorflow-feedstock/pull/62

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @conda-forge/tensorflow I'm building recipe for the object_detection API by following the steps on `tensorflow-feedstock` recipe. please help us understand the issue with skipping test for `linux` and `osx`.

File: recipes/tf_object_detection/meta.yaml (hunk 4 above)
Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @scopatz on the recipe for [tensorflow-feedstock](https://github.com/conda-forge/tensorflow-feedstock/blob/master/recipe/meta.yaml), testing on `linux` and `OSX` has been is indicated to be challenging. The comments section says:

       Skip the import test on Linux as wheel file require a more recent
       version of GLIBC++ than the VM used to build and test package.
       Also skip on OSX due to OSX version requirements, see
       https://github.com/conda-forge/tensorflow-feedstock/pull/62

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @conda-forge/tensorflow I'm building recipe for the object_detection API by following the steps on `tensorflow-feedstock` recipe. please help us understand the issue with skipping test for `linux` and `osx`.

File: recipes/tf_object_detection/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{% set name = "tf_object_detection" %}
+{% set version = "0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+source:
+  url: https://github.com/Maxar-Corp/ratpac/releases/download/od-{{ version }}/{{ name }}-{{ version }}.tar.gz
```
Comment (hajapy) [inline recipes/tf_object_detection/meta.yaml]: I’m not comfortable with this source: there is no code documenting how this tarball is built, and it is not the official source. I’d much prefer a recipe that pulls from the official source and documents the build steps to turn that into a conda package as part of the conda recipe dir.

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @hajapy it would be great the pull the tar.gz from the source it self but the [object detection API](https://github.com/tensorflow/models/tree/master/research/object_detection) is still a folder with a bunch of scripts and not as installable software. I w'd rather have the team at tensorflow package and release the tar file but as you can see they haven't. Thats why I moved the folder to a place where the team I work with can actively support the files.

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: as far as` a code documenting how this tarball is built`, I'm happy to do that. If you have suggestion on how to do this better, please let me know.

Comment (hajapy) [reply recipes/tf_object_detection/meta.yaml]: I think the value of getting this into CF outweighs waiting for when/if google ever turns the objdet api into a pip package, so I support your effort here. I'm hoping we can find a better way to do that such that the code visibly comes from the official source and the mechanism by which it turns into a conda package is codified in the recipe.

1. All the steps you followed to build the tarball in the first place should be converted into a build.sh and added to this recipe. (I imagine this involves: contributing a setup.py so it can be packaged, building the protos, including the slim package somehow). The setup.py can be added using a patch.

2. The source url should point to https://github.com/tensorflow/models, though you may want to point to a recent sha off of master rather than the outdated tags.

3. Versioning will be weird. Since there are no pip packages to match, you could do our own versioning (perhaps starting at 0.0.1). 


A brief search also brought up some other efforts to address this:

This fork adds a setup.py and was released to pypi. Not official, nor up to date. 
https://github.com/brandonschabell/models/tree/master/research/object_detection
https://pypi.org/project/object-detection

An issue requesting this feature: https://github.com/tensorflow/models/issues/4887 and a corresponding PR exemplifying how to convert into pip installable distribution: https://github.com/tensorflow/models/pull/4889

File: recipes/tf_object_detection/meta.yaml (hunk 4 above)
Comment (jjhelmus) [reply recipes/tf_object_detection/meta.yaml]: The `tensorflow` packages on conda-forge are repackage versions of the PyPI wheels provided by Google.  Despite being labeled as `manylinux1` these wheels do not meet the manylinux specification.  Specifically the wheels and conda packages require Ubunutu 16.04 and must be run on a processor with the AVX instruction set.  Therefore they will not work on the CentOS 6 image that is used for building and testing on conda-forge.  More details on this can be found in conda-forge/tensorflow-feedstock#6.  Building Tensorflow from source would produce packages which are compatible with CentOS 6 but this process is very time consuming, see conda-forge/tensorflow-feedstock#69, if anyone wants to work on this help would be appreciated.

File: recipes/tf_object_detection/meta.yaml (hunk 5 above)
Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @hajapy great. I will continue to explore this: writing up the `built.sh` and tagging `sha` off of master.

File: recipes/tf_object_detection/meta.yaml (hunk 4 above)
Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @jjhelmus thanks. I hope that will help @scopatz understand the issue here.

File: recipes/tf_object_detection/meta.yaml (hunk 5 above)
Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @asimshankar or @karmel would you help the community bring TF_object_detection to Conda-Forge. It w'd be challenging without release of an official tarball and if you can assist in any way, it w'd be greatly appreciated. There was some discussion on creating a `pip` installable file [last year](https://github.com/tensorflow/models/issues/4887) but is no progress since. I'll be glad to assist in any way possible.

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @hajapy hope all is good. can you explain what you mean by "contributing a setup.py so it can be packaged"? correct me if I'm wrong but using a patch for the setup.py w'd require an PR accepted by Google so that setup.py.patch file indexed and add the setup file using `git -- diff setup.py`. any thoughts?

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: @scopatz. I hope the thread answered some of the question you raised in reviewing this. 

@hajapy asked the source url of this PR be directly connected to google rather than what I have on my github but it seems that tensorflow developers are not going to release a tarball anytime soon. 
I reached out to some and looked into previous PRs on Tensorflow and they don't seem to be interested in packaging the object detection and slim API. I'm happy to move the source when they release it but in the mean time, can I push a PR where the source is connected to my github. any thoughts?

Comment (scopatz) [reply recipes/tf_object_detection/meta.yaml]: Are there objections to merging this?

Comment (adbeda) [reply recipes/tf_object_detection/meta.yaml]: The user `Andrew Ashkar` raised his concern on the source url. But he mentioned that he understand the challenge here. Google doesn't release the tarballs. 

@conda-forge/tensorflow Is there any objection to creating a PR for Object detection and Slim off of a personal account (at least for now) since Google doesn't (won't) release tarballs for these files? Please see the thread for detail

**Comment by adbeda** [comment]
@scopatz I couldn't locate this package on either under the feedstock or the package'ss list? any reason this hasn't showed yet?

**Comment by adbeda** [comment]
@conda-forge/core I couldn't locate this package on the feedstock although it was merged two days ago. can you help figure out whats going on?

**Comment by mbargull** [comment]
You removed the recipe in https://github.com/conda-forge/staged-recipes/pull/8717/commits , https://github.com/conda-forge/staged-recipes/pull/8717/commits/497a4eb5dd0ba00771c41c57e75d8e3a9dd59395 to be exact, before this PR has been merged into `master`.
Since gh-8717 has been merged earlier, the Git history already included the addition as well as removal of this PR. Hence, the merge of this PR was a no-operation as `master` already included all commits.
(Yes, this can be confusing at first and possibly seem time travel related ;) )

Long story short: Please resubmit this PR based on current `master`.
In general, you should always open PRs from a new branch off of the newest upstream `master` and never remove recipes from this repository in any commit.

**Comment by adbeda** [comment]
This PR is not correctly merged. It is replaced with #8800
